### PR TITLE
Support for forcing the mode (auto, browser or node) (fix #102).

### LINKED ISSUE #102: Problems with isInBrowser() and jsdom
I use jsdom for testing some jQuery functionality using node and when installing source-map-support, I get the problem, that the isInBrowser() function returns true, though I'm really only faking it. And then it can't load the source maps.

I have two suggestions for fixing this: 

1. Only check for browser-capabilities on construction and let isInBrowser be constant. That way I could simply just install jsdom after source-map-support. And I can't think of any other way isInBrowser could change from call to call.

2. Make an option on install to force either node or browser behavior.

I would be happy to make a PR if any of the above solutions seems ok to you :)

And great project by the way!

### PR DESCRIPTION
**Pull request by tugend**
Hi there, I had a talk with Asger and thought I might give this a try. =P (see 

I have added an option 'environment' which can be either 'auto', 'node' or 'browser'.

Cheers =)

### CODE REVIEW

File: source-map-support.js [hunk 1]
```diff
@@ -403,9 +412,13 @@ exports.install = function(options) {
     options = options || {};
     var installHandler = 'handleUncaughtExceptions' in options ?
       options.handleUncaughtExceptions : true;
+      
     emptyCacheBetweenOperations = 'emptyCacheBetweenOperations' in options ?
       options.emptyCacheBetweenOperations : false;
 
+    environment = 'environment' in options ?
```
Comment (julien-f) [inline source-map-support.js]: An `if` would be enough since you set the default value line 12.

File: source-map-support.js [hunk 2]
```diff
@@ -18,7 +21,13 @@ var sourceMapCache = {};
 var reSourceMap = /^data:application\/json[^,]+base64,/;
 
 function isInBrowser() {
-  return ((typeof window !== 'undefined') && (typeof XMLHttpRequest === 'function'));
+  if (environment === "browser")
+    return true;
+  if (environment === "node")
+    return false;
+  if (environment === "auto")
```
Comment (julien-f) [inline source-map-support.js]: No `if` needed there since you already check for the other 2 possible values.

File: source-map-support.js [hunk 3]
```diff
@@ -18,7 +21,13 @@ var sourceMapCache = {};
...
+    return ((typeof window !== 'undefined') && (typeof XMLHttpRequest === 'function'));
+  throw new Error("environment " + environment + " was unknown. Available options are {auto, browser, node}")
```
Comment (julien-f) [inline source-map-support.js]: I think this test would be better in `install()`.

**Comment by julien-f** [comment]
Rebase on `master`, it should fix the tests.

**Comment by tugend** [comment]
@julien-f, thanks for the feedback.
I updated the pullrequest, as before let me know if you spot room for improvement. 

Cheers.

**Comment by julien-f** [comment]
It would be good to have tests for this feature but I don't know an easy way to do that :/

**Comment by julien-f** [comment]
If no one knows how to properly test this feature, I propose we go ahead and merge it.

@tugend Could you squash your commits and have a better commit message (I think the PR's title I just set is clearer).

**Comment by tugend** [comment]
Whoops. Manged to close the pullrequest by playing with rebase and squash. 
It's the same pullrequest but with a new id: https://github.com/evanw/node-source-map-support/pull/111

**Pull request closed without merging**